### PR TITLE
docs: update header docs to actually work

### LINKED ISSUE #3835: docs: Custom Headers `token` example produces TypeScript error
### Area of Improvement

In the [v.10 docs section on Custom Headers](https://trpc.io/docs/header), the following example is suggested for sending an authorization header from the client:

```typescript
// Import the router type from your server file
import type { AppRouter } from '@/server/routers/app';
import { httpBatchLink } from '@trpc/client';
import { createTRPCNext } from '@trpc/next';

export let token: string;

export const trpc = createTRPCNext<AppRouter>({
  config({ ctx }) {
    return {
      links: [
        httpBatchLink({
          url: 'http://localhost:3000/api/trpc',
          /**
           * Headers will be called on each request.
           */
          headers() {
            return {
              Authorization: token,
            };
          },
        }),
      ],
    };
  },
});
```

It is also suggested that `token` should be imported in another source file, and modified appropriately, as in the following code snippet (taken from the same docs and section)
```typescript
// MY ADDITION: There's an implicit: import { token, trpc } from "utils/trpc.ts"
const loginMut = trpc.auth.login.useMutation({
  onSuccess({ accessToken }) {
    token = accessToken;
  },
});
```
However, when I try to follow the above pattern which involves `import { token } from "utils/trpc.ts"`, I end up with a TypeScript error `Cannot assign to 'token' because it is an import`. So it seems to be a read-only reference. To get around this issue, I exported an object from `utils/trpc.ts`, and then would modify a field on the object, as such:
```typescript
export const headers = { authorization: "" };
```

and then I would be able to modify the `headers.authorization` field in any other source file, as desired. This workaround allows us to modify the authorization header sent from the client.

### Link to related docs

https://trpc.io/docs/header

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing the suggested changes!

### PR DESCRIPTION
**Pull request by iduuck**
Closes #3835

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?
- Changed [documentation for custom headers](https://trpc.io/docs/header)
- Added new example for custom headers

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: examples/minimal-react-with-custom-headers/client/src/Greeting.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react';
+import { setToken } from './utils/client';
+import { trpc } from './utils/trpc';
+
+export function Greeting() {
+  const { data, refetch } = trpc.greeting.useQuery();
+
+  const onUpdateToken = useCallback(() => {
+    console.log('onUpdateToken')
+    setToken("custom-token")
```
[the remark this answers is not in the document]
Comment (iduuck) [reply examples/minimal-react-with-custom-headers/client/src/Greeting.tsx]: That's fine.

**Comment by iduuck** [comment]
I honestly don't know, though, why the new example is crashing on CI.

File: examples/minimal-react-with-custom-headers/client/src/Greeting.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useCallback } from 'react';
+import { setToken } from './utils/client';
+import { trpc } from './utils/trpc';
+
+export function Greeting() {
+  const { data, refetch } = trpc.greeting.useQuery();
+
+  const onUpdateToken = useCallback(() => {
+    console.log('onUpdateToken');
+    setToken('custom-token');
```
[the remark this answers is not in the document]
Comment (iduuck) [reply examples/minimal-react-with-custom-headers/client/src/Greeting.tsx:10]: all good here.

**Review by KATT — COMMENTED** [review]
@iduuck can we get rid of the example? It feels a bit noisy to have an example just to show custom header functionality

**Comment by iduuck** [comment]
sure, just wanted to have an example for the issue owner

**Comment by KATT** [comment]
friendly ping @iduuck :)

**Comment by iduuck** [comment]
hey, @KATT. had a ton to do at work, will get to it today.

**Comment by iduuck** [comment]
merged upstream, and removed example folder, @KATT

File: .github/workflows/examples.yml [hunk 3]
```diff
@@ -36,6 +36,7 @@ jobs:
             .interop/next-prisma-starter,
             next-minimal-starter,
             minimal-react,
+            minimal-react-with-custom-headers,
```
Comment (juliusmarminge) [inline .github/workflows/examples.yml:39]: rm

**Review by juliusmarminge — COMMENTED** [review]
Why is there a diff in the pnpm-lock?

**Comment by iduuck** [comment]
restored everything but the docs change

**Review by ixahmedxi — APPROVED** [review]

**Pull request merged**